src/common-ssh/common-ssh/key.h

```diff
@@ -25,75 +25,23 @@
 #include <guacamole/client.h>
 #include <libssh2.h>
 
-#include <openssl/ossl_typ.h>
-
-/**
- * The expected header of RSA private keys.
+/* OpenSSH v1 private keys are PEM-wrapped base64-encoded blobs. The encoded data begins with:
+ *   "openssh-key-v1\0"
+ * and continues with the following data for a plaintext key:
+ *   pascal string: cipher name ("none")
+ *   pascal string: kdf name ("none")
+ *   pascal string: kdf params (NULL)
+ *   32-bit int: number of keys (1)
  */
-#define SSH_RSA_KEY_HEADER "-----BEGIN RSA PRIVATE KEY-----"
+#define OPENSSH_V1_KEY_HEADER "-----BEGIN OPENSSH PRIVATE KEY-----\nb3BlbnNzaC1rZXktdjEA"
 
-/**
- * The expected header of DSA private keys.
- */
-#define SSH_DSA_KEY_HEADER "-----BEGIN DSA PRIVATE KEY-----"
-
-/**
- * The size of single number within a DSA signature, in bytes.
- */
-#define DSA_SIG_NUMBER_SIZE 20
-
-/**
- * The size of a DSA signature, in bytes.
- */
-#define DSA_SIG_SIZE DSA_SIG_NUMBER_SIZE*2 
-
-/**
- * The type of an SSH key.
- */
-typedef enum guac_common_ssh_key_type {
-
-    /**
-     * RSA key.
-     */
-    SSH_KEY_RSA,
-
-    /**
-     * DSA key.
-     */
-    SSH_KEY_DSA
-
-} guac_common_ssh_key_type;
+#define OPENSSH_V1_UNENCRYPTED_KEY "AAAABG5vbmUAAAAEbm9uZQAAAAAAAAAB"
```

Comment:
The contextual comment at above is helpful, but please document each of these constants separately such that they are individually clear. For example:

https://github.com/apache/guacamole-server/blob/90322cd4d3797edb01706b2d327c7d85a26cd31f/src/common-ssh/common-ssh/key.h#L30-L33

and

https://github.com/apache/guacamole-server/blob/90322cd4d3797edb01706b2d327c7d85a26cd31f/src/common-ssh/common-ssh/key.h#L35-L38

Reply:
Done